### PR TITLE
PSQLADM-511: proxysql-admin syncusers hangs at 1000 DB user or above

### PR DESCRIPTION
**Pull request by venkatesh-prasad-v**
https://perconadev.atlassian.net/browse/PSQLADM-511

Problem
-------
Logic used by --syncusers is of O(n^2). When it tries to create a new user in proxysql, it compares the new user to be added against all the existing users thereby resulting in a huge delay.

Solution
--------
Modified the logic of --syncusers to use assosciative arrays to perform comparison of MySQL and ProxySQL users.

The old implementation took about 6 mins to sync 500 users and 51 mins to sync 1000 users.

The new implementation takes about 12 seconds to sync 500 users and 25 seconds to sync 1000 users.

In addition to the above,
- Moved the syncusers() function to proxysql-admin-common so that the same logic will be used both by the proxysql-admin and
percona-scheduler-admin scripts.

- Introduces a new bats test to test the new improvement.

### CODE REVIEW

**Comment by venkatesh-prasad-v** [comment]
Test Results: https://gist.github.com/venkatesh-prasad-v/618837e61f1101fe5dea01306bd419aa

File: proxysql-admin-common [hunk 1]
```diff
@@ -517,6 +559,8 @@ function syncusers() {
     echo -e "\nSyncing user accounts from server(${CLUSTER_HOSTNAME}:${CLUSTER_PORT}) to ProxySQL"
 
   else
+    cluster_in_proxysql_check $WRITER_HOSTGROUP_ID
```
Comment (kamil-holubicki) [inline proxysql-admin-common]: As a good practice use {} always `${WRITER_HOSTGROUP_ID}`

**Review by kamil-holubicki — COMMENTED** [review]
Running `shellcheck` reports a lot of warnings.

File: proxysql-admin-common (hunk 1 above)
Comment (venkatesh-prasad-v) [reply proxysql-admin-common]: Agreed!

Comment (venkatesh-prasad-v) [reply proxysql-admin-common]: Done

**Comment by venkatesh-prasad-v** [comment]
@kamil-holubicki The errors reported by the shellcheck utility tool have been there from the beginning. I've created https://perconadev.atlassian.net/browse/PSQLADM-513 to fix it in the next release.

File: proxysql-admin-common [hunk 2]
```diff
@@ -574,131 +618,158 @@ function syncusers() {
...
+
+    # Add ProxySQL user to hash
+    proxysql_users_full_hash["${username}"]="$password"
+
+  done < <(printf "%s\n" "${proxysql_users}")
+
+  unset username
+  unset password
+
+  # Disable nounset to be able to search in the assiciative array
```
Comment (kamil-holubicki) [inline proxysql-admin-common]: `assiciative` typo

**Review by kamil-holubicki — APPROVED** [review]

**Pull request merged**